includes/optional-modules/class-nextdoor.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * Nextdoor integration for Newspack.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use Newspack\Nextdoor\API;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Nextdoor integration class.
+ */
+class Nextdoor {
+
+	/**
+	 * Settings slug.
+	 *
+	 * @var string
+	 */
+	const SETTINGS_SLUG = 'newspack_nextdoor_settings';
+
+	/**
+	 * Capability slug for publishing to Nextdoor.
+	 *
+	 * @var string
+	 */
+	const CAPABILITY_SLUG = 'np_nextdoor_publish_posts';
+
+	/**
+	 * Initialize the module.
+	 */
+	public static function init() {
+		// Only initialize if the module is active.
+		if ( ! Optional_Modules::is_optional_module_active( 'nextdoor' ) ) {
+			return;
+		}
+
+		// Add custom capability.
+		add_action( 'admin_init', [ __CLASS__, 'add_nextdoor_capability' ] );
+
+		// Enqueue post editor scripts.
+		add_action( 'enqueue_block_editor_assets', [ __CLASS__, 'enqueue_post_editor_assets' ] );
+
+		// Include required files.
+		require_once NEWSPACK_ABSPATH . 'includes/optional-modules/nextdoor/class-api.php';
+		require_once NEWSPACK_ABSPATH . 'includes/optional-modules/nextdoor/class-auth.php';
+		require_once NEWSPACK_ABSPATH . 'includes/optional-modules/nextdoor/class-controller.php';
+	}
+
+	/**
+	 * Enqueue post editor assets.
+	 */
+	public static function enqueue_post_editor_assets() {
+		// Only load on post editor for users who can publish to Nextdoor.
+		$screen     = get_current_screen();
+		$post_types = apply_filters( 'newspack_nextdoor_publish_cap_post_types', [ 'post' ] );
+		if ( ! $screen || ! in_array( $screen->post_type, $post_types, true ) || ! self::can_user_publish() ) {
+			return;
+		}
+
+		$handle = 'newspack-nextdoor-post-editor-plugin';
+		wp_enqueue_script(
+			$handle,
+			\Newspack\Newspack::plugin_url() . '/dist/other-scripts/nextdoor.js',
+			[ 'wp-edit-post', 'wp-data', 'wp-components', 'wp-element' ],
+			NEWSPACK_PLUGIN_VERSION,
+			true
+		);
+
+		wp_enqueue_style(
+			$handle,
+			\Newspack\Newspack::plugin_url() . '/dist/other-scripts/nextdoor.css',
+			[],
+			NEWSPACK_PLUGIN_VERSION
+		);
+	}
+
+	/**
+	 * Add custom Nextdoor capability to appropriate roles.
+	 */
+	public static function add_nextdoor_capability() {
+		$roles_with_cap = self::get_nextdoor_capability_roles();
+
+		foreach ( $roles_with_cap as $role_name ) {
+			$role = get_role( $role_name );
+			if ( $role && ! $role->has_cap( self::CAPABILITY_SLUG ) ) {
+				$role->add_cap( self::CAPABILITY_SLUG );
+			}
+		}
+	}
+
+	/**
+	 * Get roles that have Nextdoor publishing capability.
+	 *
+	 * @return array
+	 */
+	public static function get_nextdoor_capability_roles() {
+		$settings       = self::get_settings();
+		$roles_with_cap = isset( $settings['allowed_roles'] ) ? $settings['allowed_roles'] : [ 'administrator' ];
```

Comment:
```suggestion
		$roles_with_cap = isset( $settings['allowed_roles'] ) ? (array) $settings['allowed_roles'] : [ 'administrator' ];
```

Just in case there's some weird saved data :)

Reply:
Good catch! added casting ✔️